### PR TITLE
Check i18n ymls

### PR DESCRIPTION
**Pull request by evansiroky**
This PR introduces a new `scripts` package with the following:
- `validate-i18n`: a script to ensure that i18n message IDs that exist in the code also exist in the i18n language files and vice versa. This script is invoked with `yarn test` during the CI build.
- `collect-i18n-messages` extracts all messages for one or multiple languages into a CSV file (for folks who prefer using spreadsheets).

To test locally, try commenting out a few strings from one or all of the language files, run `yarn test`, and observe the resulting error messages.

### CODE REVIEW

**Review by miles-grant-ibigroup — COMMENTED** [review]
Some of the `forEach`s could maybe be refactored into reducers, which might allow some code re-use, but when I thought about it I don't think it would really improve readability that much.

This works great and everything that I had a question about was commented. Glad we finally got this up and running!

File: packages/scripts/package.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{
+  "name": "@opentripplanner/scripts",
```
[the remark this answers is not in the document]
Comment (binh-dam-ibigroup) [reply packages/scripts/package.json:2]: Not so sure, the script to update dependencies is a "private" script, whereas the ones in the scripts package are intended to be public (so we can use them in OTP-RR, for instance).

File: packages/scripts/src/collect-i18n-messages.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+async function collectAndPrintOutMessages({ sourceFiles, ymlFilesByLocale }) {
+  // Gather message ids from code.
+  const messagesFromCode = JSON.parse(await extract(sourceFiles, {}));
+  const messageIdsFromCode = Object.keys(messagesFromCode);
+  const allLocales = Object.keys(ymlFilesByLocale);
+
+  // CSV heading
+  console.log(`ID,Description,${allLocales.join(",")}`);
+
+  // Will contain id, description, and a column for each language.
```
[the remark this answers is not in the document]
Comment (binh-dam-ibigroup) [reply packages/scripts/src/collect-i18n-messages.ts:37]: Good idea, added in https://github.com/opentripplanner/otp-ui/pull/287/commits/6a391d8b1588bd1b3b1d5bb801153dcbc417a5c3.

File: packages/scripts/src/collect-i18n-messages.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        allI18nMessagesFlattened = {
+          ...allI18nMessagesFlattened,
+          ...flattenedMessages
+        };
+      });
+
+      messageIdsFromCode.filter(isNotSpecialId).forEach(id => {
+        const { description } = messagesFromCode[id];
+        const message = allI18nMessagesFlattened[id]?.trim() || undefined;
+        // console.log(`${id},"${description}","${message}"`);
```
[the remark this answers is not in the document]
Comment (binh-dam-ibigroup) [reply packages/scripts/src/collect-i18n-messages.ts]: Yep, removed in https://github.com/opentripplanner/otp-ui/pull/287/commits/6a391d8b1588bd1b3b1d5bb801153dcbc417a5c3

File: packages/scripts/src/util.ts [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+import path from "path";
+
+export interface SourceFilesAndYmlFilesByLocale {
+  sourceFiles: string[];
+  ymlFilesByLocale: Record<string, string>;
+}
+
+function shouldProcessFile(fileName: string): boolean {
+  return (
+    !fileName.includes("*") &&
```
[the remark this answers is not in the document]
Comment (binh-dam-ibigroup) [reply packages/scripts/src/util.ts]: Thanks, removed in https://github.com/opentripplanner/otp-ui/pull/287/commits/6a391d8b1588bd1b3b1d5bb801153dcbc417a5c3, that was from before when processing command line directly.

File: packages/scripts/package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (miles-grant-ibigroup) [reply packages/scripts/package.json:2]: Makes sense

**Review by miles-grant-ibigroup — APPROVED** [review]
Looks excellent thanks for the changes!

**Review by daniel-heppner-ibigroup — APPROVED** [review]
Nice utility, thank you!

**Pull request merged**